### ISSUE TITLE
docker-compose setup instructions outdated/incomplete

### ISSUE THREAD

**Issue by gbrgr** [finos/legend#474]
### Steps to Reproduce:
 1. Follow the instructions stated in the [docker-compose setup guide with a remote Gitlab instance](https://github.com/finos/legend/blob/master/installers/docker-compose/legend-with-remote-gitlab/README.md)

### Expected Result:
Working run of Legend Studio via `http://localhost:8080/studio`

### Actual Result:

 It seems that the applications ignores the configured public host ip address when redirecting (as stated in the configuration parameter `HOST_ADDRESS`, as I always experience an auth call that redirects to `localhost`. One can solve this issue by visiting `http://<your-local-ip>/studio` instead of using `localhost`. 

Edit: Maybe this has to to with how docker networking works under MacOS. I haven't tried it on a Linux machine.

### Environment:
Run on MacOS 12.0.1
Docker version 20.10.10, build b485636
Docker Compose version v2.1.1

**Comment by beekemarie** [comment]
Thanks @gbrgr! @epsstan @maoo - would you mind taking a look at this please?

**Comment by akphi** [comment]
@gbrgr Hi, I believe this is a limitation that we have already highlighted. I forgot the details, but there's something wrong with our redirection setup and the way docker containers communicate. So I think we already highlighted the point you mentioned in the `README` - https://github.com/finos/legend/blob/master/installers/docker-compose/legend-with-remote-gitlab/README.md#docker-compose-for-legend-with-remote-instance-of-gitlab (see step 4)

A bit more about our redirection, when you go to Studio, we will call SDLC to check for your authorization status, if you're not authorized, we will call SDLC to authorize you, also sending it the current url for Studio. This means if you visit Studio using `localhost`, the redirection will also be `localhost` in SDLC, and at this point the limitation with `docker compose` I mentioned before kicked in, causing SDLC to fail - I couldn't quite fully recollect what we found out at the time, but this is the version I remember.
